Test/Integration/Model/Order/HydrateOrderFromQuoteTest.php

```diff
@@ -132,15 +132,7 @@ public function testHydratesOrderFromQuoteSuccessfully(bool $quoteItemHasWeight)
             $this->addWeightToFirstQuoteItem($quote);
         }
 
-        $result = $hydratedOrderFromQuote->hydrate($quote, $publicOrderId);
-        $expectedResultBody = [
-            'status' => 201,
-            'errors' => [],
-            'body' => []
-        ];
-
-        self::assertSame(201, $result->getStatus());
-        self::assertEquals($expectedResultBody, $result->getBody());
+        $hydratedOrderFromQuote->hydrate($quote, $publicOrderId);
```

Comment:
Can we add after `$this->expectNotToPerformAssertions();` or `self:expectNotToPerformAssertions();` so we explicitly let anyone reading it later that we only expect it to not throw exception for the scenario?

Reply:
Technically, there _is_ an assertion in this test: the mock for the Bold API Client expects that it will receive a given data array and output a specific result body. I don't normally like to write mocks like this because they tie the test to to its subject's implementation, but for this code, there was no way around that due to the way it was written. (I normally only use mocks as stubs with no expectations.) To make this code easier to test, the logic for generating the request body _should_ be in its own class that could be tested separately and then stubbed out in this test. We can probably revisit that refactoring in the future, though.

CC: @nicolenorman